cpp/include/cudf/detail/iterator.cuh

```diff
@@ -80,6 +80,79 @@ struct null_replaced_value_accessor
   }
 };
 
```

Comment:
I notice that you have an accessor that replaces the value of null elements, and one that only works for non-nullable elements. What about an accessor that just returns the value and validity of elements in a nullable column without replacing them? Is this not needed?

Reply:
Element at null position, is an invalid data. I don't see any use case where they would want read this. 
I will update `make_pair_iterator` for this as well.

Reply:
added

Reply:
If it's not needed, why did you add it? I was only asking why so I can understand the usage.